contributors/devel/sig-instrumentation/migration-to-structured-logging.md

```diff
@@ -214,37 +220,85 @@ func LogHTTP(r *http.Request) {
 }
 ```
 
-Names of arguments should use [lowerCamelCase] and be alphanumeric. Arguments names in one log call should be unique.
-Names should be picked based on semantic meaning of value itself, not the context in which is used (log message should
-imply the context). For example names like `status` should be used over (`desiredStatus`, `oldStatus`, `badStatus`) thus
-allowing to query and join different log lines of the `status` field.
-
-Kubernetes objects should be referenced using only their kind, no matter their api group or version. Example argument
-names: `deployment`, `pod`, `node`, `replicaSet`. For objects of unknown type, is ok to log them under `object` key with
-addition of `apiVersion` and `kind` fields describing the k8s object type.
-
-In situations when we want to the log value of the same meaning twice (e.g. transition between state) it is ok to name
-an additional argument based on context, but leaving one most current/correct value with canonical name.
-
-Examples of keys (strongly suggested, will be extended when pattern emerge, no standard schema yet):
-* `err` - error when using `klog.InfoS`. Used for expected errors that are not `klog.ErrorS`.
-* `object` - reference to k8s objects of unknown type. Should be used with `kind` and `apiVersion`.
-* `kind` - kind of k8s object of unknown type.
-* `apiVersion` -  API version of k8s object of unknown type.
-
-Example:
+When deciding on names of arguments you should:
+* Always use [lowerCamelCase], for example use `containerName` and not `container name` or `container_name`.
+* Use [alphanumeric] characters and not should not any special characters like `%$*`, non-latin or unicode characters.
+* Use object kind when referencing Kubernetes objects, for example `deployment`, `pod` and `node`.
+* Describe the type of value stored under they key and avoid to:
+  * Include any of value properties, for example instead of `directory` use `path`.
+  * Describe context how value is used, for example instead of `podIP` use `IP`.
+  * Shorten the name (except acronyms), for example instead of `addr` use `address`
+  * Use ambiguous names, that require falling back on message to undestand what value it is. For example instead of
+    `key` use `cacheKey` or instead of `version` use `dockerVersion`.
+* Be consistent, for example when logging file path we should always use `path` and not switch between
+  `hostPath`, `path`, `file`.
+
+Exemptions (some cases are temporary workarounds that may change if we settle on better solution):
+* `err` is one case where short for error should be used, reason is to match key used by `klog.ErrorS`
+* Context in name is acceptable to distinguish between values that normally go under same key. For example using both
+  `status` and `oldStatus` in log that needs to show the change between statuses.
```

Comment:
Shouldn't the other status be `newStatus` then?

Reply:
I expect that logs referring to state change will be much rarer than ones describing the state at the moment of writing logs. New state after the change could be considered the current state at the moment of writing logs, so instead of creating new group of keys prefixed with "new", we can use the fact it's the current state for log allowing for correlation with other logs.